### PR TITLE
First steps of the hovering feature and simple test case

### PR DESCRIPTION
**Pull request by Danyboyyy**
Hovering is now working with a hard-coded string, this is just to make sure that the server can handle requests properly. Also created a simple test case to make sure it works properly.

### CODE REVIEW

File: src/EditorFeatures/Core/ExternalAccess/EditorConfig/Features/EditorConfigQuickInfo.cs
Comment (CyrusNajmabadi) [inline src/EditorFeatures/Core/ExternalAccess/EditorConfig/Features/EditorConfigQuickInfo.cs:17]: nit: IEnumerables as fields can be risky.  it is not clear if they're lazy or not and if random work spikes may happen each time they are used.

File: src/EditorFeatures/Core/ExternalAccess/EditorConfig/Features/EditorConfigQuickInfo.cs
Comment (CyrusNajmabadi) [inline src/EditorFeatures/Core/ExternalAccess/EditorConfig/Features/EditorConfigQuickInfo.cs:13]: nit: why do we need a new type for this?  are existing quick info types not suitable?

File: src/EditorFeatures/Core/ExternalAccess/EditorConfig/Features/HoverHandler.cs
Comment (CyrusNajmabadi) [inline src/EditorFeatures/Core/ExternalAccess/EditorConfig/Features/HoverHandler.cs:38]: @dibarbet @Danyboyyy do we need the lsp solution?  if so, waht for?

File: src/EditorFeatures/Core/ExternalAccess/EditorConfig/Features/IEditorConfigQuickInfoService.cs
Comment (CyrusNajmabadi) [inline src/EditorFeatures/Core/ExternalAccess/EditorConfig/Features/IEditorConfigQuickInfoService.cs:17]: surprised we're not making a subclass of QuickInfoService for the editorconfig case.  note: we would need to revise that API to support TextDocument, not just Document.  But i think that would be totally clean to do.

File: src/Features/LanguageServer/Protocol/Workspaces/LspWorkspaceManager.cs
Comment (CyrusNajmabadi) [inline src/Features/LanguageServer/Protocol/Workspaces/LspWorkspaceManager.cs:243]: surprising how complex this is.  if we're doing the same for normal documents, can we somehow share all this logic?

**Comment by CyrusNajmabadi** [comment]
Done with pass.

File: src/EditorFeatures/Core/ExternalAccess/EditorConfig/Features/EditorConfigQuickInfo.cs
Comment (Danyboyyy) [reply src/EditorFeatures/Core/ExternalAccess/EditorConfig/Features/EditorConfigQuickInfo.cs:13]: My bad, all of the QuickInfo stuff wasn't supposed to be on this PR 😅

File: src/EditorFeatures/Core/ExternalAccess/EditorConfig/Features/HoverHandler.cs
Comment (dibarbet) [reply src/EditorFeatures/Core/ExternalAccess/EditorConfig/Features/HoverHandler.cs:38]: yeah - the editorconfig file is part of the LSP solution

File: src/EditorFeatures/Core/ExternalAccess/EditorConfig/Features/EditorConfigQuickInfo.cs
Comment (dibarbet) [inline src/EditorFeatures/Core/ExternalAccess/EditorConfig/Features/EditorConfigQuickInfo.cs:13]: this looks like it isn't used yet right?  if so we should leave it out of this pr (and may not need it entirely)

File: src/EditorFeatures/Core/ExternalAccess/EditorConfig/Features/IEditorConfigQuickInfoService.cs
Comment (dibarbet) [reply src/EditorFeatures/Core/ExternalAccess/EditorConfig/Features/IEditorConfigQuickInfoService.cs:17]: yeah this is certainly one way we could do this.  But lets put this in the next PR (since it looks its unused currently).

File: src/EditorFeatures/TestUtilities/LanguageServer/AbstractLanguageServerProtocolTests.cs
Comment (dibarbet) [inline src/EditorFeatures/TestUtilities/LanguageServer/AbstractLanguageServerProtocolTests.cs:427]: I think actually you can just concat the workspace.Documents and works.AnalyzerConfigDocuments and then do GetRequiredTextDocument on either of them instead of copying the same impl - https://sourceroslyn.io/#Microsoft.CodeAnalysis.Workspaces/ISolutionExtensions.cs,64

File: src/Features/LanguageServer/Protocol/Handler/RequestContext.cs
Comment (dibarbet) [inline src/Features/LanguageServer/Protocol/Handler/RequestContext.cs:62]: At some point we might want to have the handler define a RequestContext<T> that tells us whether the handler is looking for a regular document, additional document, analyzer config document, etc.  But not blocking this PR.

File: src/Features/LanguageServer/Protocol/Workspaces/LspWorkspaceManager.cs
Comment (dibarbet) [reply src/Features/LanguageServer/Protocol/Workspaces/LspWorkspaceManager.cs:243]: this should be shareable - after all a `Document` is a `TextDocument`.  We should be able to share everything except for where we get the documents from the solution (`lspSolution.GetAnalyzerConfigDocuments(uri);` vs `var documents = lspSolution.GetDocuments(uri);`).

Likely we should have a helper method that takes in the function to get the documents from the solution
e.g.
`GetLspDocuments<T>(TextDocumentIdentifier textDocumentIdentifier, Func<T, Solution> documentGetter, CancellationToken cancellationToken) where T : TextDocument`

File: src/Features/LanguageServer/Protocol/Extensions/Extensions.cs
Comment (dibarbet) [inline src/Features/LanguageServer/Protocol/Extensions/Extensions.cs:139]: This can just be a cast, e.g. (T)solution.GetRequiredTextDocument...

File: src/Features/LanguageServer/Protocol/Workspaces/LspWorkspaceManager.cs
Comment (dibarbet) [inline src/Features/LanguageServer/Protocol/Workspaces/LspWorkspaceManager.cs:211]: GetLspDocumentAsync should call into this with just Document

File: src/Features/LanguageServer/Protocol/Workspaces/LspWorkspaceManager.cs
Comment (dibarbet) [inline src/Features/LanguageServer/Protocol/Workspaces/LspWorkspaceManager.cs:191]: nit - rename to textDocuments

File: src/Features/LanguageServer/Protocol/Workspaces/LspWorkspaceManager.cs
Comment (dibarbet) [inline src/Features/LanguageServer/Protocol/Workspaces/LspWorkspaceManager.cs:202]: FindTextDocumentInProjectContext returns a non-nullable value, so this should just be a cast

File: src/Features/LanguageServer/Protocol/Workspaces/LspWorkspaceManager.cs
Comment (dibarbet) [inline src/Features/LanguageServer/Protocol/Workspaces/LspWorkspaceManager.cs:212]: the code here to add the document to miscellaneous files should only run for the `Document` version of `GetLspDocumentAsync` - thats because the misc files workspace only supports real documents at the moment.  So I think these two lines just need to move into that method

**Review by dibarbet — APPROVED** [review]

**Comment by dibarbet** [comment]
@Danyboyyy looks like there are a couple of style warnings to fix before I can merge (ignore the integration-corehost failures)

**Pull request merged**
